src/videoframeprocessor/DefaultVideoTransformDevice.ts

```diff
@@ -0,0 +1,221 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import DefaultBrowserBehavior from '../browserbehavior/DefaultBrowserBehavior';
+import Device from '../devicecontroller/Device';
+import VideoTransformDevice from '../devicecontroller/VideoTransformDevice';
+import Logger from '../logger/Logger';
+import DefaultVideoFrameProcessorPipeline from './DefaultVideoFrameProcessorPipeline';
+import DefaultVideoTransformDeviceObserver from './DefaultVideoTransformDeviceObserver';
+import VideoFrameProcessor from './VideoFrameProcessor';
+import VideoFrameProcessorPipeline from './VideoFrameProcessorPipeline';
+import VideoFrameProcessorPipelineObserver from './VideoFrameProcessorPipelineObserver';
+
+/**
+ * [[DefaultVideoTransformDevice]] is an augmented [[VideoInputDevice]].
+ * It transform the input {@link Device} with an array of {@link VideoFrameProcessor} to produce a `MediaStream`.
+ */
+export default class DefaultVideoTransformDevice
+  implements VideoTransformDevice, VideoFrameProcessorPipelineObserver {
+  pipe: DefaultVideoFrameProcessorPipeline | null = null;
```

Comment:
Should this be VideoFrameProcessorPipeline?

Reply:
make sense, will change.